plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryIntegrationSmokeTest.java

```diff
@@ -317,6 +317,26 @@ public void testRepeatCountAggregationView()
         onBigQuery("DROP VIEW " + viewName);
     }
 
+    /**
+     * https://github.com/trinodb/trino/issues/8183
+     */
+    @Test
+    public void testColumnPositionMismatch()
+    {
+        String tableName = "test.test_column_position_mismatch";
+
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        assertUpdate(format("CREATE TABLE %s (c_varchar VARCHAR, c_int INT, c_date DATE)", tableName));
+        onBigQuery(format("INSERT INTO %s VALUES ('a', 1, '2021-01-01')", tableName));
+
+        // Adding a CAST makes BigQuery return columns in a different order
```

Comment:
The CAST is not pushed down into the connector (it's not "visible" to the connector).
What was triggering the original problem?

(i have some doubts about https://github.com/trinodb/trino/issues/8170#issuecomment-855324032)

Reply:
@findepi To be more precise, `GenericRecord record` is returned in the fixed order `c_varchar VARCHAR, c_int INT, c_date DATE` regardless of CAST, but `List<BigQueryColumnHandle> columns` order differ if CAST exists. 

Without CAST (varchar -> int -> date)
<img width="862" alt="Screen Shot 2021-06-07 at 18 46 10" src="https://user-images.githubusercontent.com/6237050/120996480-4f6b9c00-c7c1-11eb-89c4-4ecbcee3af9d.png">

With CAST (varchar -> date -> int)
<img width="869" alt="Screen Shot 2021-06-07 at 18 45 28" src="https://user-images.githubusercontent.com/6237050/120996507-57c3d700-c7c1-11eb-87f3-606219e36fa2.png">

Do you have new suggested comment? How about `// Adding a CAST changes the column order in BigQuery page source`?

Reply:
> Adding a CAST changes the column order in BigQuery page source

This is true currently, but may not be true tomorrow.

This is related to https://github.com/trinodb/trino/pull/7756 cc @losipiuk .
I don't think we can reliably test that. **IF** we can, we should test this for all connectors.

Anyway, i am sure CAST was not a necessary condition to trigger the bug, so we should update the release notes entry (eg make more vague).

Reply:
@findepi  Updated as below. Please feel free to edit my comment. 
`Fix incorrect results when accessing BigQuery records with wrong index.`